### PR TITLE
Update changelog.md

### PR DESCRIPTION
**Pull request by amyblais**
Adding notes about upcoming dot releases to make users aware.

### CODE REVIEW

File: source/administration/changelog.md [hunk 1]
```diff
@@ -77,6 +83,11 @@ Mattermost v5.15.0 contains low level security fixes. [Upgrading](http://docs.ma
 
 ## Release v5.14 - [Feature Release](https://docs.mattermost.com/process/release-faq.html#release-overview)
 
+- **v5.14.5, released 2019-10-16**
```
Comment (lieut-data) [inline source/administration/changelog.md]: Will we actually have a `v5.14.5` or merging the two together? The order here seems off as well if we've released one before the other.

Comment (amyblais) [reply source/administration/changelog.md]: @lieut-data The notification issue is caused by the upcoming v5.16 / desktop v4.3 release so the backports for that issue won't be released before October 16th.

The two issues you're working on are existing issues so I'm assuming we'll have a dot release for those before v5.16 release?

Comment (amyblais) [reply source/administration/changelog.md]: Probably need clarification from @esethna if we can release the notification issue backports before October 16th.

Comment (lieut-data) [reply source/administration/changelog.md]: Ah, that makes sense!

Comment (esethna) [reply source/administration/changelog.md]: Checking with dev. I think we may be able to release them earlier than v4.3 assuming there are no compatibility issues with notifications if using the backported server versions with v4.2

Comment (amyblais) [reply source/administration/changelog.md]: From Dean's notes, it seems it's ok to have all of these in the same dot release, so I made that change here. The changelog can be edited later if anything changes

**Pull request merged**